### PR TITLE
Fix segmentation fault in ParallelConnectedComponents

### PR DESCRIPTION
**Pull request by Relux-the-Relux**
this addresses #942 

It was previously just initializing everything to the `none` which caused just for the label `none` to be passed around in the algorithm, not assigning anything. So now all nodes are now assigned to their own partition. As none of the nodes were being assigned it would fail the `assert(component[u] != none);` from `ComponentDecomposition::componentOfNode`.

Also, the partition needed to be compacted at the end because the partition IDs would not be continuous and the number of partitions is not updated. Non-existing node entries also need to be removed from the partition in the case where the Graph node Ids are not continuous so that the number of components is correct, otherwise, they would be seen as single node components(the vectors for those partitions would be empty in the result from `getComponents()`).

### CODE REVIEW

**Comment by Relux-the-Relux** [comment]
> Why does this fix the issue? Can you add a comment? Why is compaction of component IDs needed here?

updated my comment with the explanation.

**Comment by avdgrinten** [comment]
Where (and why) do we get a segfault if there are unused node IDs in the partition?

File: networkit/cpp/components/test/ConnectedComponentsGTest.cpp [hunk 1]
```diff
@@ -93,6 +93,22 @@ TEST_F(ConnectedComponentsGTest, testParallelConnectedComponents) {
...
+    G.addEdge(1, 2);
+    G.addEdge(3, 4);
+    G.addEdge(4, 5);
+    ConnectedComponents cc(G);
+    cc.run();
+    count seqNum = cc.numberOfComponents();
+    ParallelConnectedComponents ccPar(G);
+    ccPar.run();
+    count parNum = ccPar.numberOfComponents();
+    EXPECT_EQ(seqNum, parNum);
```
Comment (angriman) [inline networkit/cpp/components/test/ConnectedComponentsGTest.cpp:108]:
```suggestion
    ParallelConnectedComponents ccPar(G);
    ccPar.run();
    EXPECT_EQ( cc.numberOfComponents(), ccPar.numberOfComponents());
```

Could you also:
- add another test with a bigger graph from `input`
- check that the components computed by `ParallelConnectedComponents` in the next test are the same as those computed by `ConnectedComponents` (as you are already doing below).

**Comment by Relux-the-Relux** [comment]
I think #942 failed to correctly transcribe what they actually got., for when testing I also only got 1 parallel component.

**Comment by Relux-the-Relux** [comment]
I think #942 simply mistakenly wrote the wrong number of PCC that they got. Might be a typo. Since I also only observed 1 PCC.

**Comment by academe-01** [comment]
> I think #942 simply mistakenly wrote the wrong number of PCC that they got. Might be a typo. Since I also only observed 1 PCC.

wasn't a typo, everything been copy/pasted from interpreter

File: networkit/cpp/components/test/ConnectedComponentsGTest.cpp [hunk 2]
```diff
@@ -90,6 +90,12 @@ TEST_F(ConnectedComponentsGTest, testParallelConnectedComponents) {
         count parNum = cc.numberOfComponents();
         DEBUG("Number of components: ", seqNum);
         EXPECT_EQ(seqNum, parNum);
+        G.forNodes([&](node u) { EXPECT_EQ(cc.componentOfNode(u), ccPar.componentOfNode(u)); });
+        ccPar.runSequential();
+        parNum = cc.numberOfComponents();
```
Comment (fabratu) [inline networkit/cpp/components/test/ConnectedComponentsGTest.cpp:95]: `parNum = ccPar.numberOfComponents();`

File: CMakeLists.txt [hunk 3]
```diff
@@ -486,10 +486,15 @@ if (NETWORKIT_BUILD_TESTS)
 					networkit_state
 			)
 		endif()
-
+		
+		if(WIN32)
+			set(NETWORKIT_TESTS_CXX_FLAGS "${NETWORKIT_CXX_FLAGS} /wd4996")
```
Comment (fabratu) [inline CMakeLists.txt:491]: Only minor, but no need to introduce a new variable. Just reuse `NETWORKIT_CXX_FLAGS`. We do so in other parts of the CMake-file.

**Review by fabratu — APPROVED** [review]
LGTM now.

**Pull request merged**